### PR TITLE
Prevent leaking password reset token through Referrer header

### LINKED ISSUE #28: Leaking of password reset token through the reset url
I think it is possible to leak the password reset token since it is left in the url. 

In Django 1.11 the token is stripped during a redirect ([docs](https://docs.djangoproject.com/en/1.11/releases/1.11/#django-contrib-auth), [code](https://github.com/django/django/blob/1.11.2/django/contrib/auth/views.py#L461)) to prevent the token from being taken in the referrer header from 3rd party apps on the page. I haven't dug too deeply into the source for this project but at a first glance it seems that the vulnerability exists. 

If this is the case would be happy to help fix similar to [django-registration-redux](https://github.com/macropin/django-registration/pull/268), or at the very least alert you to the issue. Let me know if you guys need any help!

### PR DESCRIPTION
**Pull request by joshblum**
Fixes #28

### CODE REVIEW

File: users/urls.py [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
-                                       password_reset_confirm,
-                                       password_reset_done)
+from django.contrib.auth import views as auth_views
 
 from .views import (activate, activation_complete, register,
                     registration_closed, registration_complete)
 
+# Attempt to use the auth class based views if available.
+try:
+    login_view = auth_views.LoginView.as_view()
```
Comment (mishbahr) [inline users/urls.py]: AFAIK this won't work. e.g. `LoginView.as_view()` need to called w/ `template_name` as kwarg.

Same issue w/ all the views being called without kwags. 

I would prefer that we simply use the class based views.. and add a warning to README for folks to pin their requirements to use version 0.2.1 or earlier for older django.

**Comment by mishbahr** [comment]
Thanks for your pull request. Please see cooments.

**Comment by joshblum** [comment]
@mishbahr let me know if you need any other changes!

File: README.rst [hunk 2]
```diff
@@ -21,8 +21,9 @@ django-users2
   :target: https://coveralls.io/r/mishbahr/django-users2?branch=master
 
 
-Custom user model for django >=1.5 with support for multiple user types and
-lots of other awesome utils (mostly borrowed from other projects).
+Custom user model for django >=1.11 with support for multiple user types and
```
Comment (mishbahr) [inline README.rst]: Please can we change the wording to something like this:

> Custom user model for django >=1.5 with support for multiple user types and lots of other awesome utils (mostly borrowed from other projects). If you are using django < 1.11, please install v0.2.1 or earlier (`pip install django-users2<=0.2.1`).

So it's clearer that, it still works for the older versions of django.

Comment (joshblum) [inline README.rst]: done!

**Pull request merged**